src/model.cpp

```diff
@@ -691,16 +742,29 @@ void flattenComponent(const ComponentEntityPtr &parent, const ComponentPtr &comp
             importedComponentCopy->addComponent(component->component(i));
         }
 
-        // Temporarily add component to new model to find units used.
+        // Temporarily add component to new model to find units used by the variables
         auto tempModel = Model::create();
         tempModel->addComponent(importedComponentCopy);
         tempModel->linkUnits();
+
         std::vector<UnitsPtr> requiredUnits;
         for (size_t i = 0; i < tempModel->unitsCount(); ++i) {
             auto u = tempModel->units(i);
             requiredUnits.push_back(u);
         }
 
+        // Get the names of the units used by the <cn> elements in all the components' mathml and add them
+        // to the required units
```

Comment:
Missing full stop at the end of the sentence. Also, "MathML" not "mathml".

Reply:
Added fullstop.  We use "mathml" throughout the code too...  should fix all of them separately?

Reply:
Done, also changed other occurrences.